vowpalwabbit/learner.h

```diff
@@ -186,18 +228,44 @@ struct learner
   func_data end_pass_fd;
   func_data end_examples_fd;
   func_data finisher_fd;
-
   std::shared_ptr<void> learner_data;
+
   learner(){};  // Should only be able to construct a learner through init_learner function
+
  public:
   prediction_type_t pred_type;
   size_t weights;  // this stores the number of "weight vectors" required by the learner.
   size_t increment;
   bool is_multiline;  // Is this a single-line or multi-line reduction?
+  std::string name;   // Name of the reduction.  Used in VW_DBG to trace nested learn() and predict() calls
```

Comment:
my comment on example_predict.h applies here too, I find it concerning that debug only build is now enforcing requirements on abstractions. I might be too strict though.

Reply:
Yes I understand the general object to having this.  I responded to you earlier comments.  This is a point of contention.  We should discuss during the VW meeting.